src/js/common/schemaform/review/ReviewChapters.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import ReviewCollapsibleChapter from './ReviewCollapsibleChapter';
+import {
+  createPageListByChapter,
+  expandArrayPages,
+  getActiveChapters,
+  getPageKeys
+} from '../helpers';
+import {
+  getReviewPageOpenChapters,
+  getViewedPages
+} from '../state/selectors';
+import { getActivePages } from '../../../../platform/forms/helpers';
+import {
+  closeReviewChapter,
+  openReviewChapter,
+  setData,
+  setEditMode,
+  setViewedPages,
+  uploadFile
+} from '../actions';
+import Scroll from 'react-scroll';
+
+const scroller = Scroll.scroller;
+class ReviewChapters extends React.Component {
+
+  componentDidMount() {
+    const pageList = this.props.pageList;
+    const form = this.props.form;
+
+    this.props.setViewedPages(new Set(getPageKeys(pageList, form)));
+  }
+
+  handleToggleChapter({ name, open, pageKeys }) {
+    if (open) {
+      this.props.closeReviewChapter(name, pageKeys);
```

Comment:
Chapter `pageKeys` are passed to the action creator and added to the view state. 
![viewedpages](https://user-images.githubusercontent.com/4998130/39340958-34ff1cde-498e-11e8-9511-b0796c178594.gif)